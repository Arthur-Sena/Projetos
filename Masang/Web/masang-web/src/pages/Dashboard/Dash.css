:root {
    --background: #f2f2f2;
    --laranja-claro: #f28123;
    --vermelho: #B90404;
    --text-color: #868686;
  }

.Dashboard {
    background-color: var(--background);
    height: 100vh;
}

/* -------Div do Grafico------- */
.tituloGrafico{
    background-image: linear-gradient(180deg, var(--laranja-claro), var(--vermelho));
    color: white;
    display:flex;
    flex-direction: column;
    justify-content: center;
    height: 7vh;
    border-top-left-radius: 15px;
    border-top-right-radius: 15px;
    border-style: solid;
    border-width: 1px;
    border-color: var(--vermelho);
}

.h3titulo{
    text-align: center;
    color: white;
}

.divGrafico{
    margin-top: 10vh;
}

.grafico{
    background-color: white;
    margin-left: 5vw;
    height: 73vh;
    width:47vw;
    border-color: var(--vermelho);
    border-style: solid;
    border-width: 1px;
    border-bottom-left-radius: 15px;
    border-bottom-right-radius: 15px;
    /* position: absolute;
    margin-top: -52vh; */
}

.Dashboard{
    display:flex;
    flex-direction: row;
}

.listaDash{
    margin-left: 2vw;
}

.divsQnt{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.userTitle{
    background-image: linear-gradient(180deg, var(--laranja-claro), var(--vermelho));
    width: 19.5vw;
    height: 7vh;
    margin-top: 10vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
    border-top-left-radius: 15px;
    border-top-right-radius: 15px;
    border-style: solid;
    border-width: 1px;
    border-color: var(--vermelho);
}

.numero{
    background-color: white;
    color: var(--text-color);
    width: 19.5vw;
    height: 7vh;
    border-style: solid;
    border-width: 1px;
    border-color: var(--vermelho);
    display: flex;
    flex-direction: column;
    justify-content: center;
    text-align: center;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
}

.listaDoacoes{
    margin-top: 5vh;
    display:flex;
    flex-direction: column;
}
.ultimasDoacoes{
    height: 52.5vh;
    margin-top: 0.5vh;
    direction: rtl;
    text-align: left;
    overflow: auto;
    margin-left: 10px;
    padding-right: 8px;
}
.infoDoacao{
    margin-left: 10px;
}

/* Barra de Rolamento */
.ultimasDoacoes::-webkit-scrollbar{
    width: 8px;
}
.ultimasDoacoes::-webkit-scrollbar-thumb         
{
    background-color: rgba(136,136,136,0.4);
}
.bottomList{
    height: 3px;
    width: 30vw;
    background-image: linear-gradient(to right , var(--laranja-claro), var(--vermelho));
    margin-right: 5vw;
}

.testandoDiv{
    height: 54vh;
    width:40vw;
    background-color: white;
    border-color: var(--vermelho);
    border-style: solid;
    border-width: 1px;
    border-bottom-left-radius: 15px;
    border-bottom-right-radius: 15px;
}
