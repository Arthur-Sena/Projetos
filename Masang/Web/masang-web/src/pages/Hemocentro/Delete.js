import React, { Component } from 'react';
import Axios from 'axios';

import Loading from '../../assets/Loading.gif'

class Del extends Component {

    deletarHemocentro = () => {
        Axios.delete('https://masang.dev.objects.universum.blue/hemocentros/' + localStorage.getItem('Xid-HX'))
            .then(() =>  localStorage.removeItem('Xid-HX') + this.props.history.push('/Mapa'))
    }

    componentDidMount() {
        this.deletarHemocentro();
    }

    render() {
        return (
            <div className="DivDelete"  
            style={{
                backgroundColor:"#f2f2f2" ,
                minHeight: '100vh',
                display:'flex',
                flexDirection: 'column',
                justifyContent: 'center',
                alignItems:'center'
                }}  >
                <img src={Loading} style={{ height: '20vh'}}/>
                <h3  style={{color: '#f29123'}}>Deletando Hemocentro</h3>
            </div>
        );
    }
}
export default Del