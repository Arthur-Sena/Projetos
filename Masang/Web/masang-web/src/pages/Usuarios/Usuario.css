.divUser {
    background-color: var(--background);
    height: 105vh;
}

.divTable {
    
    display: flex;
    justify-content: center;
    margin-top: 3vh;
}

.userTable {
    width: 80vw;
    text-align: center;
    border-color: var(--vermelho);
    border-style: solid;

    background-color: white;

}

.inputTable {
    margin-top: 7vh;
    display: flex;
    justify-content: center;
}

.thead{
    background-image: linear-gradient(180deg, var(--laranja-claro), var(--vermelho));
}

.tbody{
    color: var(--text-grey);
}
.tr{
    height: 5vh;
}
.th{
    height: 6vh;
}

/* --------------------------------------------------------------------------- */
.divOneUser{
    height: 101vh;
    background-color: var(--background);
    
}

.infoUser{
    background-color: white;
    
    margin-left: 15vw;
    margin-top: 1vh;

    width: 65vw;
    height: 65vh;
    border-radius: 50px;
    display: flex;
    flex-direction: row;
    /* justify-content: center; */
    padding: 3vw;

    border-color: grey;
    border-style: solid;
    border-width: 1px;
}

.imgUser{
    height:65vh;
    width:20vw;
    border-radius: 20px;
    overflow: hidden;
    border-color: #b90404;
    border-style: solid;
    border-width: 2px;
}

.dadosUser{
    padding-left: 6vw;
    padding-top: 3vh;
    color: var(--laranja-claro);
    font-size: larger;
    font-weight: 500;
    display: flex;
    flex-direction: column;
    /* justify-content: space-between; */
}

