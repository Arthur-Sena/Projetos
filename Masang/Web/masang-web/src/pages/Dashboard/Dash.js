import React, { Component } from 'react';
import { Route, Link, BrowserRouter as Router, Switch, Redirect } from "react-router-dom";
import Chart from '../../components/Chart';


// // import Axios from 'axios';

// //Css + images
import './Dash.css';
import Logo from '../../assets/logoMasangofc1.png';

let ap = [];
let an = [];
let bp = [];
let bn = [];
let op = [];
let on = [];
let Abp = [];
let Abn = [];

class Dashboard extends Component {

    constructor(props) {
        super(props);
        this.state = {
            chartData: {},
            qntUser: '',
            qntHemocentro: '',
            pedidos: [],
            sangueAp: 4,
            sangueAn: 2,
            sangueBp: 2,
            sangueBn: 0,
            sangueABp: 0,
            sangueABn: 0,
            sangueOp: 1,
            sangueOn: 0,
        }
    }
    // Contadores de hemocentro e usuarios
    countHemocentro = () => {
        fetch('https://masang.dev.objects.universum.blue/hemocentros/count')
            .then(response => response.json())
            .then(data => this.setState({ qntHemocentro: data })
            )
    }
    countUser = () => {
        fetch('https://masang.dev.objects.universum.blue/usuarios/count')
            .then(response => response.json())
            .then(data => this.setState({ qntUser: data })
            )
    }
    // Listar Usuarios
    listUser = () => {
        fetch('https://masang.dev.objects.universum.blue/usuarios/')
        .then(response => response.json())
        .then(data => {
            ap = data.filter( x => x.tipoSanguineo === 'A+')
            this.setState({ sangueAp : ap.length})

            an = data.filter( x => x.tipoSanguineo === 'A-')
            this.setState({ sangueAn : an.length})

            bp = data.filter( x => x.tipoSanguineo === 'B+')
            this.setState({ sangueBp : bp.length})

            bn = data.filter( x => x.tipoSanguineo === 'B-')
            this.setState({ sangueBn : bn.length})
          
            Abp = data.filter( x => x.tipoSanguineo === 'AB+')
            this.setState({ sangueABp : Abp.length})

            Abn = data.filter( x => x.tipoSanguineo === 'AB-')
            this.setState({ sangueABn : Abn.length})

            op = data.filter( x => x.tipoSanguineo === 'O+')
            this.setState({ sangueOp : op.length})
            
            on = data.filter( x => x.tipoSanguineo === 'O-')
            this.setState({ sangueOn : on.length})
        })
    }

    
    // Lista de Pedidos
    listPedidos = () => {
        fetch('https://masang.dev.objects.universum.blue/pedidos')
            .then(response => response.json())
            .then(data => this.setState({ pedidos: data.reverse() })
            )
    }

    componentDidMount() {
        this.countHemocentro();
        this.countUser();
        this.listPedidos();
        this.listUser();
    }
    componentWillMount() {
        this.getChartData();
    }
    // Grafico
    getChartData() {
        this.setState({
            chartData: {
                labels: ['A+', 'A-', 'B+', 'B-', 'AB+', 'AB-', 'O+', 'O-'],
                datasets: [
                    {
                        label: 'Quantidade de Usuarios',
                        data: [ this.state.sangueAp , this.state.sangueAn , this.state.sangueBp , this.state.sangueBn,
                        this.state.sangueABp, this.state.sangueABn, this.state.sangueOp, this.state.sangueOn],
                        backgroundColor: [
                            'rgba(255, 99, 132, 0.6)',
                            'rgba(99, 255, 132, 0.6)',
                            'rgba(99, 132, 255, 0.6)',
                            'rgba(132, 99, 255, 0.6)',
                            'rgba(255, 132, 99, 0.6)',
                            'rgba(132, 255, 99, 0.6)',
                            'rgba(255, 0, 255, 0.6)',
                            'rgba(99, 99, 99, 0.6)'
                        ],
                    }
                ]
            }
        })
    }
    render() {
        return (
            <div className="Dashboard" >
                {/* Grafico */}
                <div className="divGrafico">
                    <div className="tituloGrafico" style={{ marginLeft: "5vw", width: '47vw' }}>
                        <h3 className="h3titulo">Quantidade de doadores por tipo sanguíneo</h3>
                    </div>
                    <div className="grafico">
                        <Chart chartData={this.state.chartData} />

                    </div>
                </div>

                <div className="listaDash">
                    {/* Quantidade de Hemocentro e Usuarios */}
                    <div className="divsQnt">
                        <div className="userQnt">
                            <div className="userTitle">
                                {/* <Link to="/Usuarios" style={{ textDecoration: 'none' }}><h3 className="h3titulo"> Quantidade de Usuarios</h3></Link> */}
                                <a href="/Usuarios" style={{ textDecoration: 'none' }}><h3 className="h3titulo"> Quantidade de Usuarios</h3></a>

                            </div>
                            <div className="numero">
                                <a href="/Usuarios" style={{ textDecoration: 'none', color: '#868686' }}><h3>{this.state.qntUser}</h3></a>
                            </div>
                        </div>
                        <div className="userQnt" style={{ marginLeft: '1vw' }}>
                            <div className="userTitle">
                                {/* <Link to="/Mapa" style={{ textDecoration: 'none' }}><h3 className="h3titulo"> Quantidade de Hemocentros</h3></Link> */}
                                <a href="/Mapa" style={{ textDecoration: 'none' }}><h3 className="h3titulo"> Quantidade de Hemocentros</h3></a>

                            </div>
                            <div className="numero">
                                {/* <Link to="/Mapa" style={{ textDecoration: 'none', color: '#868686' }}><h3>{this.state.qntHemocentro}</h3></Link> */}
                                <a href="/Mapa" style={{ textDecoration: 'none', color: '#868686' }}><h3>{this.state.qntHemocentro}</h3></a>

                            </div>
                        </div>
                    </div>

                    <div className="listaDoacoes" >
                        <div className="tituloGrafico" style={{ width: '40vw' }}>
                            <h3 className="h3titulo">Ultimas Pedidos de Doação</h3>
                        </div>
                        <div className="testandoDiv">
                            <div className="ultimasDoacoes">
                                {this.state.pedidos.map(element => {
                                    let efetuado;
                                    let x = element.dataSolicitacao;
                                    let y = x.split(' ');
                                    let z = y[0].split('-');
                                    let data = z[2] + "/" + z[1] + "/" + z[0];

                                    element.efetuadoMatch === 'true' ? (efetuado = "já encontrou doador  ") : (efetuado = "ainda não encontrou doardor")
                                    return (
                                        <div className="infoDoacao">
                                            <p style={{ fontSize: 'large', fontWeight: "500", color: '#868686' }}>Pedido de sangue {element.tipoSanguineo} cadastrado no dia {data} , {efetuado}</p>
                                            <div className="bottomList" style={{ marginTop: '-7px' }}></div>
                                            {/* <h4>An: {this.state.sangueAn}, Ap= {this.state.sangueAp}, Bp={this.state.sangueBp}, Bn={this.state.SangueBn}, ABp={this.state.sangueABp}</h4> */}
                                        </div>
                                    );
                                })}
                            </div>
                        </div>
                    </div>
                </div>

            </div>
        );
    }
}


export default Dashboard

