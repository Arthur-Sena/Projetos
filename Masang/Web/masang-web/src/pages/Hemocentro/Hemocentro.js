import React, { Component } from 'react';
import { Route, Link, BrowserRouter as Router, Switch, Redirect } from "react-router-dom";

import Axios from 'axios';

//Css + images
import './Hemocentro.css';

class Hemocentro extends Component {

  constructor(props) {
    super(props);
    this.state = {
      dados: {},
      local: {},

      nome: "",
      tel: "",
      horarioatendimento: "",
      longitude: "",
      latitude: "",
      cep: "",
      estado: '',
      cidade: '',
      rua: '',
      numero: '',
      sang1: '',
      sang2: '',
      sang3: '',
      sang4: '',
      sang5: '',
      sang6: '',
      sang7: '',
      sang8: '',

      erro: '',
    }
  }

  buscarHemocentro = () => {
    fetch('https://masang.dev.objects.universum.blue/hemocentros/' + localStorage.getItem('Xid-HX'))
      .then(response => response.json())
      .then(data => {
        this.setState({ dados: data })
        this.setState({ nome: data.nome })
        this.setState({ telefone: data.telefone })

        let x = JSON.parse(data.endereco)
        this.setState({ cidade: x.cidade })
        this.setState({ estado: x.estado })
        this.setState({ numero: x.numero })
        this.setState({ rua: x.rua })
        this.setState({ cep: x.cep })
        this.setState({ horarioatendimento: data.horarioatendimento })

        let y = JSON.parse(data.localizacao)
        this.setState({ latitude: y.latitude })
        this.setState({ longitude: y.longitude })

        // let z = JSON.parse(data.tiposEmNecessidade)
      })


  }

  setNome = (event) => {
    this.setState({ nome: event.target.value })
  }
  setTel = (event) => { this.setState({ telefone: event.target.value }) }
  setCep = (event) => {
    this.setState({ cep: event.target.value })
  }
  setCidade = (event) => {
    this.setState({ cidade: event.target.value })
  }
  setEstado = (event) => {
    this.setState({ estado: event.target.value })
  }
  setEndereco = (event) => {
    this.setState({ rua: event.target.value })
  }
  setNumero = (event) => {
    this.setState({ numero: event.target.value })
  }
  setHora = (event) => {
    this.setState({ horarioAtendimento: event.target.value })
  }
  setLongitude = (event) => {
    this.setState({ longitude: event.target.value })
  }
  setLatitude = (event) => {
    this.setState({ latitude: event.target.value })
  }


  atualizarHemocentro = (event) => {
    event.preventDefault();

    Axios.put('https://masang.dev.objects.universum.blue/hemocentros/' + localStorage.getItem('Xid-HX'), {
        nome: this.state.nome,
        telefone: this.state.telefone,
        horarioatendimento: this.state.horarioAtendimento,
        localizacao: { latitude: this.state.latitude, longitude: this.state.longitude },
        endereco: {
          estado: this.state.estado,
          cidade: this.state.cidade,
          rua: this.state.rua,
          cep: this.state.cep,
          numero: this.state.numero,
        } 
      },{
      headers: {
        'Content-Type': 'application/json'
      }
    })
    .then(response => {
      if (response.status === 200) {
          alert("Hemocentro Atualizado" )
          window.location.reload();
      } else {
          this.setState({ erro: "Erro ao Atualizar" })
      }
  })
  }
// tiposEmNecessidade: [
        //   this.state.sang1,
        //   this.state.sang2,
        //   this.state.sang3,
        //   this.state.sang4,
        //   this.state.sang5,
        //   this.state.sang6,
        //   this.state.sang7,
        //   this.state.sang8
        // ]

  componentDidMount() {
    this.buscarHemocentro();
  }

  render() {

    return (
      <div className="Hemocentro" >
        <div className="hemocentroTitle">
          {/* <Link to="/Mapa" style={{ textDecorationColor: '#b90404' }}><h1 className="Title">{this.state.dados.nome}</h1></Link> */}
          <a href="/Mapa" style={{ textDecorationColor: '#b90404' }}><h1 className="Title">{this.state.dados.nome}</h1></a>

          {/* <div className="bottomTitle" style={{ width: "31vw" }}></div> */}
        </div>

        <div className="divFormCadastro" style={{ marginTop: '-2vh' }}>

          <form onSubmit={this.atualizarHemocentro} className="formCadastro">


            {/* INPUT NOME DO HEMOCENTRO --- OK*/}
            <div className="itemC" >
              <label htmlFor="nome" className='labelCadastro'>Nome do Hemocentro</label>
              <input
                className="input_cadastro"
                type="text"
                name="name"
                onChange={this.setNome}
                defaultValue={this.state.nome}
              />
            </div>



            {/* INPUT TELEFONE --- OK*/}
            <div className="itemC">
              <label htmlFor="tel" className='labelCadastro'>Telefone</label>
              <input
                className="input_cadastro"
                type="tel"
                name="telefone"
                defaultValue={this.state.telefone}
                onChange={this.setTel}
                minLength='8'
              />
            </div>



            {/* INPUT CIDADE & ESTADO --- OK*/}
            <div className="itemC1" style={{ marginTop: '-3vh' }}   >
              <div style={{ marginTop: "3vh" }}>
                <label htmlFor="cidade" className='labelCadastro'>Cidade</label>
                <input
                  className="input_cadastroA"
                  type="text"
                  name="cidade"
                  defaultValue={this.state.cidade}
                  onChange={this.setCidade}
                  minLength='3'
                  maxLength='50'
                />
              </div>
              <div className="hifen"></div>
              <div style={{ marginTop: "3vh" }}>
                <label htmlFor="nome" className='labelA'>Estado</label>
                <input
                  style={{ textAlign: 'center' }}
                  className="input_cadastroB"
                  type="text"
                  name="estado"
                  defaultValue={this.state.estado}
                  onChange={this.setEstado}
                  maxLength="3"
                // placeholder="   SP"
                />
              </div>
            </div>




            {/* INPUT CEP --- OK*/}
            <div className="itemC" style={{ marginTop: '-3vh' }}>
              <label htmlFor="cep" className='labelCadastro'  >Cep</label>
              <input
                className="input_cadastro"
                type="cep"
                name="cep"
                defaultValue={this.state.cep}

                onChange={this.setCep}
                minLength='8'
                maxLength='10'
              />
            </div>



            {/* INPUT ENDERECO --- OK*/}
            <div className="itemC1" style={{ marginTop: '-3vh' }}   >
              <div style={{ marginTop: "3vh" }}>
                <label htmlFor="cidade" className='labelCadastro'>Endereço (Av, Rua)</label>
                <input
                  // style={{width:'17vw'}}
                  className="input_cadastroA"
                  type="text"
                  name="cidade"
                  defaultValue={this.state.rua}
                  onChange={this.setEndereco}
                  minLength='3'
                />
              </div>
              <div className="hifen" style={{ marginLeft: '-5.6vw', width: '1.8vw' }}></div>
              <div style={{ marginTop: "3vh" }}>
                <label htmlFor="nome" className='labelA'>Nº</label>
                <input
                  style={{ width: '5vw', marginRight: '2.7vw', textAlign: 'center' }}
                  className="input_cadastroB"
                  type="text"
                  name="estado"
                  defaultValue={this.state.numero}
                  onChange={this.setNumero}
                  maxLength="6"
                />
              </div>
            </div>



            {/* INPUT HORA --- OK*/}
            <div className="itemC" style={{ marginTop: '-3vh' }} >
              <label htmlFor="hora" className='labelCadastro'>Horário de Atendimento</label>
              <input
                className="input_cadastro"
                type="text"
                name="hora"
                defaultValue={this.state.horarioatendimento}
                onChange={this.setHora}
              />
            </div>



            {/* INPUT LATITUDE --- OK , DA PRA MELHORAR*/}
            <div className="itemC" style={{ marginTop: '-5vh' }}>
              <label htmlFor="latitude" className='labelCadastro'>Latitude</label>
              <input
                className="input_cadastro"
                type="text"
                name="latitude"
                defaultValue={this.state.latitude}
                onChange={this.setLatitude}
                minLength="5"
                maxLength="11"
              />
            </div>



            {/* Input LONGITUDE --- OK , DA PRA MELHORAR*/}
            <div className="itemC" style={{ marginTop: '-5vh' }}>
              <label htmlFor="longitude" className='labelCadastro'>Longitude</label>
              <input
                className="input_cadastro"
                type="text"
                name="longitude"
                defaultValue={this.state.longitude}
                onChange={this.setLongitude}
                minLength="5" maxLength="11"
              />
            </div>



            {/* Sangue em necessidade  --- OK*/}
            <div className="checkBox" style={{ width: '78vw', display: 'flex', flexDirection: 'row', color: '#868686' }}>
              <label className='labelCadastro' style={{ marginTop: '-3px' }}> Sangue em Necessidade:</label>
              <div style={{ width: '45vw', display: 'flex', justifyContent: 'space-between', marginRight: '4vw' }}>
                <div style={{ marginLeft: '-7vw' }}>
                  <input onClick={this.setSangue} type="checkbox" name="gender" value="A+" />
                  <label htmlFor="tipo">A+</label>
                </div>
                <div>
                  <input onClick={this.setSangue2} type="checkbox" name="gender" value="A-" />
                  <label htmlFor="tipo">A-</label>
                </div>
                <div>
                  <input onClick={this.setSangue3} type="checkbox" name="gender" value="B+" />
                  <label htmlFor="tipo">B+</label>
                </div>
                <div>
                  <input onClick={this.setSangue4} type="checkbox" name="gender" value="B-" />
                  <label htmlFor="tipo">B-</label>
                </div>
                <div>
                  <input onClick={this.setSangue5} type="checkbox" name="gender" value="AB+" />
                  <label htmlFor="tipo">AB+</label>
                </div>
                <div>
                  <input onClick={this.setSangue6} type="checkbox" name="gender" value="AB-" />
                  <label htmlFor="tipo">AB-</label>
                </div>
                <div>
                  <input onClick={this.setSangue7} type="checkbox" name="gender" value="O+" />
                  <label htmlFor="tipo">O+</label>
                </div>
                <div >
                  <input onClick={this.setSangue8} type="checkbox" name="gender" value="O-" />
                  <label htmlFor="tipo">O-</label>
                </div>
              </div>
            </div>



            {/* Aviso de erro --- OK*/}
            <div style={{ minHeight: "3vh", marginBottom: '5px', backgroundColor: "white" }}>
              <p style={{ color: "#868686", textAlign: "center", fontSize: '20px', }}>{this.state.erro}</p>
            </div>


            {/* BTN ATUALIZAR --- DA PRA MELHORAR*/}
            <button className="btn" style={{ marginTop: "67.5vh", marginLeft: '-6vw', position: 'absolute', borderColor: 'grey' }}
              type='submit'
            >
              Atualizar
            </button >


            {/* BTN DELETAR ---  OK*/}
            <button className="btn"
              onClick={() => this.props.history.push('/Del')}
              style={{
                marginTop: "67.5vh", position: 'absolute', marginLeft: '67.5vw'
                , borderColor: 'grey', backgroundColor: '#b90404', marginLeft: '6vw'
              }}>
              Deletar
            </button >
          </form>
        </div>


      </div>
    );
  }
}


export default Hemocentro