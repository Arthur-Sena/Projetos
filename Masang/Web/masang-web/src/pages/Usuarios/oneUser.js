import React, { Component } from 'react';
import { Route, Link, BrowserRouter as Router, Switch, Redirect } from "react-router-dom";

// import Axios from 'axios';



//Css + images
import './Usuario.css';
import User from '../../assets/User.png'

let ativo;
let data = [];
class oneUser extends Component {

    constructor(props) {
        super(props);
        this.state = {
            usuario: {},
            data: [],
        }
    }

    buscarUsuario = () => {

        fetch('https://masang.dev.objects.universum.blue/usuarios/' + localStorage.getItem('Xid-UX'))
            .then(response => response.json())
            .then(data => this.setState({ usuario: data }))
            this.state.usuario.aptoADoar === 'true' ? (ativo = "Doador"):(ativo = "Não Doador")

            // let x = this.state.usuario.dataNascimento.split(' ')
            // let data = x.split('-')
        }

    componentDidMount() {
        this.buscarUsuario();
    }

    render() {
        return (
            <div className="divOneUser" >

                <div className="cadastroTitle" style={{ marginTop: '-1vh' }}>
                    {/* <Link to="/Usuarios" style={{ textDecorationColor: '#f29123' }}> */}
                    <a href="/Usuarios" style={{ textDecorationColor: '#f29123' }}>

                        <h1 className="Title"  >{this.state.usuario.nome}</h1>

                    </a>
                </div>

                <div className='infoUser'>

                    <div className='imgUser'>
                        {this.state.usuario.caminhoImagem != null ? (

                                <img src={this.state.usuario.caminhoImagem} style={{ height: '100%', width: '100%' }} />
                        ) : (
                                <img src={User} style={{ height: '100%', width: '150%', marginLeft:'-25%' }}/>
                            )}
                    </div>

                    <div className='dadosUser'>

                        <p><strong  > Nome :</strong>  {this.state.usuario.nome}</p>
                        <p><strong  > Email :</strong>  {this.state.usuario.email}</p>
                        <p><strong  > Sexo :</strong>  {this.state.usuario.sexo}</p>
                        {/* <p><strong  > Data de Nascimento :</strong>  {data[2]}/{data[1]}/{data[0]}</p> */}
                        <p><strong  > Tipo Sanguíneo :</strong>  {this.state.usuario.tipoSanguineo}</p>

                        <p><strong  > Nome :</strong>  {ativo}</p>
                    </div>



                </div>

            </div>
        );
    }
}

export default oneUser