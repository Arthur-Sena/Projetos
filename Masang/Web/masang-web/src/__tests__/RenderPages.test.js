import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';

//PÁGINAS
import CadHemocentro from '../pages/CadastroHemocentro/Cadastro';

import GetCampanhas from '../pages/Campanhas/Campanhas';
import PutCampanhas from '../pages/Campanhas/AtualizarCampanha';
import DelCampanhas from '../pages/Campanhas/DeleteCampanha';
import CadCampanhas from '../pages/Campanhas/CadastrarCampanha';

import Dashboard from '../pages/Dashboard/Dash';

import Hemocentro from '../pages/Hemocentro/Hemocentro';
import DelHemocentro from '../pages/Hemocentro/Delete';

import Usuarios from '../pages/Usuarios/Usuario';
import GetUsuario from '../pages/Usuarios/oneUser';





//TESTES INSTÂNTANEOS
describe('Cadastro Hemocentro Component [LOGINSCREEN]', () => {

    let CadHemo, GetCamp, PutCamp, DelCamp, CadCamp, Dash, Hemo, DelHemo, User, oneUser;

    beforeEach(() => {

        CadHemo = shallow(< CadHemocentro />)
        GetCamp = shallow(< GetCampanhas />)
        PutCamp = shallow(< PutCampanhas />)
        DelCamp = shallow(< DelCampanhas />)
        CadCamp = shallow(< CadCampanhas />)
        Dash = shallow(< Dashboard />)
        Hemo = shallow(< Hemocentro />)
        DelHemo = shallow(< DelHemocentro />)
        User = shallow(< Usuarios />)
        oneUser = shallow(< GetUsuario />)

    });


    test('deve executar a class e ciclos de vida sem crashar', () => {

        expect(mount(<CadHemocentro />)).toMatchSnapshot();
        expect(mount(<GetCampanhas />)).toMatchSnapshot();
        expect(mount(<PutCampanhas />)).toMatchSnapshot();
        expect(mount(<DelCampanhas />)).toMatchSnapshot();
        expect(mount(<CadCampanhas />)).toMatchSnapshot();
        expect(mount(<Dashboard />)).toMatchSnapshot();
        expect(mount(<Hemocentro />)).toMatchSnapshot();
        expect(mount(<DelHemocentro />)).toMatchSnapshot();
        expect(mount(<Usuarios />)).toMatchSnapshot();
        expect(mount(<GetUsuario />)).toMatchSnapshot();

    });

    test('deve executar o componente pai sem crashar', () => {
        expect(CadHemo).toMatchSnapshot();
        expect(GetCamp).toMatchSnapshot();
        expect(PutCamp).toMatchSnapshot();
        expect(DelCamp).toMatchSnapshot();
        expect(CadCamp).toMatchSnapshot();
        expect(Dash).toMatchSnapshot();
        expect(Hemo).toMatchSnapshot();
        expect(DelHemo).toMatchSnapshot();
        expect(User).toMatchSnapshot();
        expect(oneUser).toMatchSnapshot();
    });

});
