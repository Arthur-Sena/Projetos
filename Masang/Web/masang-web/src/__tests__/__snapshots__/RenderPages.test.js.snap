// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar a class e ciclos de vida sem crashar 1`] = `ReactWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar a class e ciclos de vida sem crashar 2`] = `ReactWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar a class e ciclos de vida sem crashar 3`] = `ReactWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar a class e ciclos de vida sem crashar 4`] = `ReactWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar a class e ciclos de vida sem crashar 5`] = `ReactWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar a class e ciclos de vida sem crashar 6`] = `ReactWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar a class e ciclos de vida sem crashar 7`] = `ReactWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar a class e ciclos de vida sem crashar 8`] = `ReactWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar a class e ciclos de vida sem crashar 9`] = `ReactWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar a class e ciclos de vida sem crashar 10`] = `ReactWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar o componente pai sem crashar 1`] = `ShallowWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar o componente pai sem crashar 2`] = `ShallowWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar o componente pai sem crashar 3`] = `ShallowWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar o componente pai sem crashar 4`] = `ShallowWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar o componente pai sem crashar 5`] = `ShallowWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar o componente pai sem crashar 6`] = `ShallowWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar o componente pai sem crashar 7`] = `ShallowWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar o componente pai sem crashar 8`] = `ShallowWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar o componente pai sem crashar 9`] = `ShallowWrapper {}`;

exports[`Cadastro Hemocentro Component [LOGINSCREEN] deve executar o componente pai sem crashar 10`] = `ShallowWrapper {}`;
